### ISSUE TITLE
Decide whether we want to use Chip or a different component

### ISSUE THREAD

**Issue by yanirs** [Automattic/abacus#164]
See https://github.com/Automattic/abacus/pull/160#discussion_r445210297

This probably also applies to `ExperimentStatus`: #148

**Comment by aaronyan** [comment]
The [Chip component](https://material-ui.com/components/chips/) sounds great! I say we also adjust the `ExperimentStatus` to use the Chip component as well.

**Comment by jessie-ross** [comment]
I would prefer not to, based solely on the style guide's recommendations:
> Usage: Chips allow users to enter information, make selections, filter content, or trigger actions.
https://material.io/components/chips#usage

This is the reasoning I see about applying the above to our project:
1. Chips are very useful components for interactivity. 
2. Good design separates interactive from non-interactive - part of the language of our design.
3. We should want to preserve them for interactive use.

I like the alternate idea we had - caption font, uppercase, and mild transparency.
RE: The ExperimentStatus in the #170: the legibility problem seems to be a bug due to double opacity being applied.

**Comment by dhurlburtusa** [comment]
If chips are intended for interactivity, then avoiding them for what we were thinking of using them for sounds good to me. What we are using them for is what many UI libraries call a Label component. I've seen it in Semantic UI. It's also what GitHub calls their components used for similar purposes. There are several "labels" on this page -- the purple Merged label and this PR's status label (currently Open).

Basically, we need a component that highlights are particular piece of textual data.

**Comment by aaronyan** [comment]
Thanks for the context and reasoning. Let's move ahead with not using a Chip component for `ExperimentStatus`. Our status will be more of a highlight of existing text, rather than an interactive button/chip.

Combining these ideas together:

> highlights are particular piece of textual data
> I like the alternate idea we had - caption font, uppercase, and mild transparency

<img width="1257" alt="Screen Shot 2020-06-29 at 5 44 55 PM" src="https://user-images.githubusercontent.com/4505888/86069965-4bbb8b00-ba30-11ea-932c-e633cdf1e7d0.png">

Also commenting here with that same information. https://github.com/Automattic/abacus/pull/170#pullrequestreview-438832710

**Comment by yanirs** [comment]
@aaronyan as noted on https://github.com/Automattic/abacus/pull/170#discussion_r447412738, combining three types of highlighting (uppercasing, background colour, font colour) in that manner seems excessive and doesn't seem to be inline with the guidelines.

Regarding Chips, the first sentence in the docs is: _"Chips are compact elements that represent an input, attribute, or action."_ I see the proposed usage in line with seeing them as attributes. However, I'm happy to have our own component for standard highlighting that we can call `Label`. But let's keep the highlighting minimal and consistent with the guidelines. It'd be similar to what you proposed for ExperimentStatus but with:

* [Black or white text colour with appropriate opacity](https://material.io/design/color/text-legibility.html#text-backgrounds)
* Background colour that is grey by default (like Chip) &ndash; can still use colours when appropriate (but not overdo it), e.g., as proposed by @aaronyan [here](https://github.com/Automattic/abacus/pull/170#discussion_r447224529)
* Lowercase or sentence-case text, depending on the context

What do you think?

In any case, let's keep this issue open until we've made a decision and applied it everywhere, as we now have Chips and the `ExperimentStatus` components on `master`.

**Comment by jessie-ross** [comment]
### Regarding the Material Docs

@yanirs, I did see that line and purposefully included only the Usage section 🐈 . This is likely an inconsistency in the documentation but even if they do include it I still think it makes sense to keep the interactive seperate from the non-interactive. 

I did think they might mean attributes as "choice" and "filter" types [judging from this section](https://material.io/components/chips#types) that sandwiches them between "input" and "action" types.

There is a "disabled" variant of a chip but it does similarly suffer from an opacity, [and seems to represent the "off" state of a chip switch.](https://material.io/components/chips#choice-chips)

### Regarding the Styling

Yanir's points regarding the styling makes sense to me: it might be too strong a highlight.
I would also add that it might distract from the rest of the row.

**Regarding this I would like to add third option to the table, (so to speak):**

> <img width="1046" alt="Screen Shot 2020-06-30 at 4 54 48 pm" src="https://user-images.githubusercontent.com/971886/86106501-33f90b00-baf3-11ea-922f-d7809f0a38f8.png">
> https://material-ui.com/store/items/devias-kit-pro/

- It is incredibly simple.
- It sufficiently highlights the status level of a row, and adds a bit of flavour to the table. 
- It doesn't use too many typographical "dimensions" like text colour, background colour and casing.
- I like the fact that it doesn't visually take over the whole row.
- Keeps nicely with the monospace, direct-display-of-code themes that are emerging for the MVP.

And we have a clear path to interactivity with chips:

<img width="396" alt="Screen Shot 2020-06-30 at 5 30 10 pm" src="https://user-images.githubusercontent.com/971886/86109834-64db3f00-baf7-11ea-86a6-48d5e39abd5f.png">

**Comment by aaronyan** [comment]
My apologies for closing the issue prematurely. Thanks for continuing the discussion.

How about we have both a `Label` component and a `Status` component? The `Label` feels too strong for the status. And the `Status` can be more simple but include color!

> I'm happy to have our own component for standard highlighting that we can call `Label`
> - Black or white text colour with appropriate opacity
> - Background colour that is grey by default (like Chip) – can still use colours when appropriate (but not overdo it)
> - Lowercase or sentence-case text, depending on the context

@yanirs' idea for the `Label` would be great for the "default" variations and "primary" metrics. Keeping these grey I think would look good.

> - It is incredibly simple.
> - It sufficiently highlights the status level of a row, and adds a bit of flavour to the table.
> - It doesn't use too many typographical "dimensions" like text colour, background colour and casing.
> - I like the fact that it doesn't visually take over the whole row.
> - Keeps nicely with the monospace, direct-display-of-code themes that are emerging for the MVP.

And @jessie-ross' idea for the `Status` would look really great in the experiment lists table. I like how the color is still there and doesn't detract from the rest of the row. The simplicity is very nice!

If we go with this style of `Status`, I think we should have the status be inside the "General` card in the details page (as opposed to next to the experiment name). We can decide this later, but wanted to run with this idea and imagine how it can be implemented on other pages.

**Comment by yanirs** [comment]
Thanks @jessie-ross and @aaronyan! Both suggestions sound good to me. Now that @dhurlburtusa added #175, we can track the `Chip` to `Label` transition there, and use #148 for the `ExperimentStatus` restyling. I edited #148 accordingly.

Now we can close this issue. :slightly_smiling_face: Great discussion! :tada:
